libvcx/src/disclosed_proof.rs

```diff
@@ -1049,6 +1053,38 @@ mod tests {
         assert_eq!(VcxStateType::VcxStateAccepted as u32, get_state(handle).unwrap());
     }
 
+    #[test]
+    #[cfg(feature = "general_test")]
+    fn test_proof_update_state_v2() {
+        let _setup = SetupStrictAriesMocks::init();
+
+        let connection_handle = connection::tests::build_test_connection();
+
+        AgencyMockDecrypted::set_next_decrypted_response(GET_MESSAGES_DECRYPTED_RESPONSE);
+        AgencyMockDecrypted::set_next_decrypted_message(mockdata_proof::ARIES_PRESENTATION_REQUEST);
+
+        let request = _get_proof_request_messages(connection_handle);
+
+        let handle = create_proof("TEST_CREDENTIAL", &request).unwrap();
+        assert_eq!(VcxStateType::VcxStateRequestReceived as u32, get_state(handle).unwrap());
+
+        generate_proof(handle, ARIES_PROVER_CREDENTIALS.to_string(), ARIES_PROVER_SELF_ATTESTED_ATTRS.to_string());
+        assert_eq!(VcxStateType::VcxStateRequestReceived as u32, get_state(handle).unwrap());
+
+        send_proof(handle, connection_handle).unwrap();
+        assert_eq!(VcxStateType::VcxStateOfferSent as u32, get_state(handle).unwrap());
+
+        ::connection::release(connection_handle);
+        let connection_handle = connection::tests::build_test_connection();
```

Comment:
It would make more sense for reader of the test to serialize connection, release it, then deserialize instead of building brand new test connection.